ownCloud/Settings/SecuritySettingsSection.swift

```diff
@@ -22,244 +22,283 @@ import ownCloudSDK
+
+		// Creation of the passcode row.
+		passcodeRow = StaticTableViewRow(switchWithAction: { (_, sender) in
+			if let passcodeSwitch = sender as? UISwitch {
+				if let viewController = self.viewController {
+
+					var passcodeViewController: PasscodeViewController?
+					var defaultMessage : String?
+
+					// Handlers
+					let cancelHandler:PasscodeViewControllerCancelHandler = { (passcodeViewController: PasscodeViewController) in
+						passcodeViewController.dismiss(animated: true, completion: {
+							self.isPasscodeSecurityEnabled = !passcodeSwitch.isOn
+						})
+						self.passcodeFromFirstStep = nil
+					}
+
+					if passcodeSwitch.isOn {
+						defaultMessage = "Enter code".localized
+					} else {
+						defaultMessage = "Delete code".localized
+					}
+
+					passcodeViewController = PasscodeViewController(cancelHandler: cancelHandler, completionHandler: { (passcodeViewController: PasscodeViewController, passcode: String) in
+						if !passcodeSwitch.isOn {
+							// Delete
+							if passcode == AppLockManager.shared.passcode {
+								// Success
+								AppLockManager.shared.passcode = nil
+								passcodeViewController.dismiss(animated: true, completion: {
+									self.isPasscodeSecurityEnabled = passcodeSwitch.isOn
+									self.updateUI()
+								})
+							} else {
+								// Error
+								passcodeViewController.message = defaultMessage
+								passcodeViewController.errorMessage = "Incorrect code".localized
+								passcodeViewController.passcode = nil
+							}
+						} else {
+							// Add
+							if self.passcodeFromFirstStep == nil {
+								// First step
+								self.passcodeFromFirstStep = passcode
+								passcodeViewController.message = "Repeat code".localized
+								passcodeViewController.passcode = nil
+							} else {
+								// Second step
+								if self.passcodeFromFirstStep == passcode {
+									// Passcode right
+									// Save to keychain
+									AppLockManager.shared.passcode = passcode
+									passcodeViewController.dismiss(animated: true, completion: {
+										self.isPasscodeSecurityEnabled = passcodeSwitch.isOn
+										self.updateUI()
+									})
+								} else {
+									//Passcode is not the same
+									passcodeViewController.message = defaultMessage
+									passcodeViewController.errorMessage = "The entered codes are different".localized
+									passcodeViewController.passcode = nil
+								}
+								self.passcodeFromFirstStep = nil
+							}
+						}
+					})
+
+					passcodeViewController?.message = defaultMessage
+					viewController.present(passcodeViewController!, animated: true, completion: nil)
+				}
+			}
+		}, title: "Passcode lock".localized, value: isPasscodeSecurityEnabled)
+
+		// Creation of the biometrical row.
+		if let biometricalSecurityName = LAContext().supportedBiometricsAuthenticationName() {
+			// Creation of the biometrical row.
```

Comment:
This is probably a leftover from another pull request but we could delete the comment at this point.

Reply:
The next commit will remove that double comment, thanks!